src/odemis/driver/andorcam3.py

```diff
@@ -858,15 +861,24 @@ def setTargetTemperature(self, temp):
             tmps_str = self.GetEnumStringImplemented(u"TemperatureControl")
             tmps = [float(t) if t is not None else 1e100 for t in tmps_str]
             if self.isWritable(u"TemperatureControl"):
+                logging.debug("Matching %g°C among: %s", temp, tmps_str)
                 tmp_idx = util.index_closest(temp, tmps)
                 self.SetEnumIndex(u"TemperatureControl", tmp_idx)
                 temp = tmps[tmp_idx]
             else:
-                temp = util.find_closest(temp, (min(tmps), 25))
-        elif self.isWritable(u"TargetSensorTemperature"):
-            self.SetFloat(u"TargetSensorTemperature", temp)
+                if temp != TEMP_NO_COOLING:
+                    temp = float(self.GetEnumStringByIndex(u"TemperatureControl", self.GetEnumIndex(u"TemperatureControl")))
+
+                logging.warning("Temperature cannot be controlled, using %g °C", temp)
+        else:  # Fallback to old (deprecated) property
+            if self.isWritable(u"TargetSensorTemperature"):
+                self.SetFloat(u"TargetSensorTemperature", temp)
+            else:
+                if temp != TEMP_NO_COOLING:
+                    temp = self.GetFloat(u"TargetSensorTemperature")
+                logging.warning("Temperature cannot be controlled, using %g °C", temp)
 
-        if temp >= 20:
+        if temp == TEMP_NO_COOLING:
```

Comment:
Should the docstring of the method be updated as well, to be clear which values are expected?

Reply:
The docstring is correct (it's a float representing a temperature). TEMP_NO_COOLING is just a special temperature which can be used to disabling the cooling.